camera-viewfinder-compose/src/main/java/com/google/jetpackcamera/viewfinder/CameraPreview.kt

```diff
@@ -15,106 +15,202 @@
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.produceState
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
-import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import com.google.jetpackcamera.viewfinder.surface.CombinedSurface
-import com.google.jetpackcamera.viewfinder.surface.CombinedSurfaceEvent
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.graphics.setFrom
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalDensity
+import com.google.jetpackcamera.viewfinder.surface.SurfaceTransformationUtil
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.asExecutor
-import kotlinx.coroutines.flow.mapNotNull
 
-private const val TAG = "Preview"
+private const val TAG = "CameraPreview"
 
 @Composable
 fun CameraPreview(
     modifier: Modifier,
     implementationMode: ImplementationMode = ImplementationMode.COMPATIBLE,
     onSurfaceProviderReady: (SurfaceProvider) -> Unit = {},
-    onRequestBitmapReady: (() -> Bitmap?) -> Unit,
+    // TODO(b/267488554): Remove setSurfaceView after implementing an alternative to
+    //  metering point factory
     setSurfaceView: (View) -> Unit
 ) {
     Log.d(TAG, "CameraPreview")
 
     val surfaceRequest by produceState<SurfaceRequest?>(initialValue = null) {
         onSurfaceProviderReady(
             SurfaceProvider { request ->
+                Log.d(TAG, "newSurfaceRequest")
                 value?.willNotProvideSurface()
                 value = request
             }
         )
     }
 
-    PreviewSurface(
+    ViewFinder(
         modifier = modifier,
         surfaceRequest = surfaceRequest,
-        setView = setSurfaceView,
-        onRequestBitmapReady = onRequestBitmapReady,
         implementationMode = implementationMode
     )
 }
 
+@SuppressLint("RestrictedApi")
 @Composable
-fun PreviewSurface(
+fun ViewFinder(
     modifier: Modifier,
     surfaceRequest: SurfaceRequest?,
-    onRequestBitmapReady: (() -> Bitmap?) -> Unit,
-    implementationMode: ImplementationMode = ImplementationMode.COMPATIBLE,
-    setView: (View) -> Unit
+    implementationMode: ImplementationMode
 ) {
     Log.d(TAG, "PreviewSurface")
 
-    var surface: Surface? by remember { mutableStateOf(null) }
+    if (surfaceRequest != null) {
+        val transformationInfo by produceState<TransformationInfo?>(
+            key1 = surfaceRequest,
+            initialValue = null
+        ) {
+            surfaceRequest!!.setTransformationInfoListener(
+                Dispatchers.Main.asExecutor()
+            ) {
+                Log.d(TAG, "TransformationInfo: $it")
+                value = it
+            }
+        }
+
+        var surface by remember { mutableStateOf<Surface?>(null) }
+        var parentViewSize by remember { mutableStateOf(Size(1, 1)) }
```

Comment:
Why start with a 1x1 size? 

If you would be using IntSize, here it would be easy to use `IntSize.Unspecified` or `IntSize.Zero`